bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/servlet/HomeConnectServlet.java

```diff
@@ -0,0 +1,621 @@
+     */
+    public void addBridgeHandler(HomeConnectBridgeHandler bridgeHandler) {
+        bridgeHandlers.add(bridgeHandler);
+    }
+
+    /**
+     * Remove Home Connect bridge handler from configuration servlet.
+     *
+     * @param bridgeHandler bridge handler
+     */
+    public void removeBridgeHandler(HomeConnectBridgeHandler bridgeHandler) {
+        bridgeHandlers.remove(bridgeHandler);
+    }
+
+    private void getRequestCountPage(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        if (bridgeHandlers.isEmpty()) {
+            getBridgesPage(request, response);
+        } else {
+            WebContext context = new WebContext(request, response, request.getServletContext());
+            context.setVariable("bridgeHandlers", bridgeHandlers);
+            templateEngine.process("requests", context, response.getWriter());
+        }
+    }
+
+    private void getApiRequestsPerSecondCsv(HttpServletResponse response, String bridgeId) throws IOException {
+        Optional<HomeConnectBridgeHandler> bridgeHandler = getBridgeHandler(bridgeId);
+        if (bridgeHandler.isPresent()) {
+            response.setContentType("text/csv");
+            PrintWriter writer = response.getWriter();
+
+            writer.println(String.format("%s,%s", "time", "requests"));
+            bridgeHandler.get().getApiClient().getLatestApiRequests().forEach(apiRequest -> writer.println(
+                    String.format("%s,%s", apiRequest.getTime().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME), 1)));
+        } else {
+            response.sendError(HttpStatus.BAD_REQUEST_400, "Unknown bridge");
+        }
+    }
+
+    private void getAppliancesPage(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        WebContext context = new WebContext(request, response, request.getServletContext());
+        context.setVariable("bridgeHandlers", bridgeHandlers);
+        templateEngine.process("appliances", context, response.getWriter());
+    }
+
+    private void processApplianceActions(HttpServletResponse response, String action, String thingId)
+            throws IOException {
+        Optional<HomeConnectBridgeHandler> bridgeHandler = getBridgeHandlerForThing(thingId);
+        Optional<AbstractHomeConnectThingHandler> thingHandler = getThingHandler(thingId);
+
+        if (bridgeHandler.isPresent() && thingHandler.isPresent()) {
+            try {
+                response.setContentType(MediaType.APPLICATION_JSON);
+                String haId = thingHandler.get().getThingHaId();
+
+                switch (action) {
+                    case ACTION_SHOW_DETAILS: {
+                        @Nullable
+                        String actionResponse = bridgeHandler.get().getApiClient().getRaw(haId,
+                                "/api/homeappliances/" + haId);
+                        response.getWriter().write(actionResponse != null ? actionResponse : "{}");
```

Comment:
You could make this a method to be able to re-use it. Then, you get also rid of the curly brackets at the `case`s.

Reply:
Unfortunately, I don't exactly understand your suggestion. Could you please give me a hand.

Reply:
Like so:
```
...
                    case ACTION_ALL_PROGRAMS: {
                        write(response, bridgeHandler, haId, "/programs");
                        break;
                    }
...
    private void write(HttpServletResponse response, Optional<HomeConnectBridgeHandler> bridgeHandler, String haId,
            String suffix)
            throws CommunicationException, AuthorizationException, ApplianceOfflineException, IOException {
        String actionResponse = bridgeHandler.get().getApiClient().getRaw(haId, "/api/homeappliances/" + haId + suffix);
        response.getWriter().write(actionResponse != null ? actionResponse : EMPTY_RESPONSE);
    }
```

Reply:
I merge it for now. You could file a follow-up PR.